docs/developer-docs/latest/development/custom-fields.md

```diff
@@ -0,0 +1,340 @@
+<!-- The section linked above will be part of another PR, but basically it's about mentioning that in schema files, custom fields are defined with `type: 'custom-field'` and a possible additional `customField` key for the uid (`global::…` or `plugin::…`), plus possibly additional keys (e.g. `options` for a `select` setting, etc.) -->
+
+::: note
+Though the recommended way to add a custom field is through creating a plugin, app-specific custom fields can also be registered within the global `bootstrap` and `register` [functions](/developer-docs/latest/setup-deployment-guides/configurations/optional/functions.md) found in `./src/index.js`.
+:::
+
+<!-- TODO: make sure this # exists in the backend custom. > models docs — will come in another PR -->
+
+<!-- ? should we document these? 👇 this was described in the [technical RFC](https://github.com/strapi/rfcs/blob/3eb034e169746558315d719ca5fb49cec854640a/rfcs/xxxx-custom-fields-api.md#motivation) but I'm unsure about what to do with it -->
+
+::: callout 🚧 Current custom fields limitations
+
+* Custom fields can not be used to register new database types in the Strapi backend.
+* Custom fields can not use the relation, component or dynamic zone data types.
+* Custom fields can only be shared using plugins.
+
+:::
+
+## Registering a custom field on the server
+
+::: prerequisites
+!!!include(developer-docs/latest/development/snippets/custom-field-requires-plugin.md)!!!
+:::
+
+Strapi's server needs to be aware of all the custom fields to ensure that an attribute using a custom field is valid.
+
+The `strapi.customFields` object exposes a `register()` method on the `Strapi` instance. This method is used to register custom fields on the server during the plugin's server [register lifecycle](/developer-docs/latest/developer-resources/plugin-api-reference/server.md#register).
+
+`strapi.customFields.register()` registers one or several custom field(s) on the server by passing an object (or an array of objects) with the following parameters:
+
+| Parameter                      | Description                                       | Type     |
+| ------------------------------ | ------------------------------------------------- | -------- |
+| `name`                         | The name of the custom field                      | `String` |
+| `plugin`<br/><br/>(_optional_) | The name of the plugin creating the custom fields<br/><br/>!!!include(developer-docs/latest/development/snippets/custom-field-plugin-optional.md)!!!          | `String` |
+| `type`                         | The data type the custom field will use           | `String` |
+
+::: note
+Currently custom fields can not add new data types to Strapi and should use existing, built-in Strapi data types (e.g. string, number, JSON — see [models documentation](/developer-docs/latest/development/backend-customization/models.md#model-attributes) for the full list).
```

Comment:
Not all Strapi types are allowed. I know you mentioned this a few paragraphs above, but someone only reading this section of the page may be confused

Reply:
I agree, will be worth repeating it here. Thank you, I'll update the content.